<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <arg name="log_path" />
    <arg name="output" />

    <arg name="robot_name" default="turtlebot" />
    <arg name="robot_prefix" default=""/>
    <arg name="frontier_selection" default="5"/>

    <node pkg="explorer" type="explorer" name="explorer" respawn="true" clear_params="true" output="$(arg output)" >

        <!-- costmaps -->
        <param name="local_costmap/robot_base_frame" value="base_link" />
        <param name="local_costmap/global_frame" value="odom" />
        <param name="robot_prefix" value="$(arg robot_prefix)"/>
        <param name="global_costmap/robot_base_frame" value="base_link" />
        <param name="global_costmap/global_frame" value="map" />
        <param name="global_costmap/map_topic" value="/map" />
        <param name="local_costmap/map_topic" value="/map" />
        <param name="local_costmap/observation_sources" value="base_scan" />
        <param name="local_costmap/base_scan/topic" value="/base_scan" />
        <param name="local_costmap/base_scan/data_type" value="LaserScan" />
        <param name="local_costmap/base_scan/marking" value="true" />
        <param name="local_costmap/base_scan/clearing" value="true" />
        <param name="local_costmap/base_scan/sensor_frame" value="base_laser_link"/>
        <param name="observation_persistence" value="0.1" />

        <!-- simulation parameters -->
        <param name="log_path" value="$(arg log_path)" />
        <param name="robot_name" value="$(arg robot_name)" />

        <!-- move_base -->
        <param name="base_link" value="base_link" />
        <param name="move_base_frame" value="map" />

        <!-- exploration parameters -->
        <param name="frontier_selection" value="$(arg frontier_selection)" />
        <rosparam file="$(find explorer)/param/pioneer3at_costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find explorer)/param/pioneer3at_costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find explorer)/param/pioneer3at_local_costmap_params.yaml" command="load" />
        <rosparam file="$(find explorer)/param/pioneer3at_global_costmap_params.yaml" command="load" />

        <remap from="/$(arg robot_name)/$(arg robot_name)/map" to="/$(arg robot_name)/map" />

    </node>

</launch>
